### PR TITLE
RFC: auto-fixes from ruff 0.6 (io.fits)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Partially address #16863 for `io.fits`

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/io/fits/fitsrec.py [hunk 1]
```diff
@@ -333,8 +333,7 @@ def from_columns(cls, columns, nrows=0, fill=False, character_as_bytes=False):
                     dim = arr.shape[0]
                 else:
                     dim = 0
-                if dim > nrows:
-                    nrows = dim
+                nrows = max(dim, nrows)
```
Comment (mhvk) [inline astropy/io/fits/fitsrec.py:336]: Fine to just stick with the simple change here, but the whole thing could be
```
if nrows == 0 and (lengths := [arr.shape[0] for arr in columns._arrays if arr is not None]):
    nrows = max(lengths)
```
(Not completely sure the guard against all arrays being `None` is necessary)

File: astropy/io/fits/header.py [hunk 2]
```diff
@@ -1423,8 +1423,7 @@ def insert(self, key, card, useblanks=True, after=False):
         # used by list.insert()
         if idx < 0:
             idx += len(self._cards) - 1
-            if idx < 0:
-                idx = 0
+            idx = max(idx, 0)
```
Comment (mhvk) [inline astropy/io/fits/header.py:1426]: Yikes both to having `+= len(self._cards) -1` (why the `-1`??) and to making everything too low 0.  But that feels very out of scope here... (without it, one could just use [`normalize_axis_index`](https://numpy.org/doc/stable/reference/generated/numpy.lib.array_utils.normalize_axis_index.html))

**Review by mhvk — APPROVED** [review]
Also OK, though both places feel ripe for more of a refactoring...

**Comment by pllim** [comment]
Hmm benchmark failures seem unrelated? 🤷‍♀️ 

```
| Change   | Before [81432542]    | After [90d70649]    |   Ratio | Benchmark (Parameter)          |
|----------|----------------------|---------------------|---------|--------------------------------|
| +        | 35.2±0.07μs          | 123±70μs            |    3.49 | units.time_quantity_creation   |
| +        | 93.2±0.5μs           | 223±100μs           |    2.39 | units.time_quantity_init_array |

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE DECREASED.
```

**Review by saimn — APPROVED** [review]
LGTM, thanks

**Comment by pllim** [comment]
@mhvk , do your concerns still stand?

**Comment by mhvk** [comment]
Here, I really wondered about the code, but since this PR is just about refactoring to use max/min, that should not stop it.

**Comment by pllim** [comment]
@mhvk , feel free to open follow-up issue if it really bothers you.

Thanks, all!

**Pull request merged**